### ISSUE TITLE
offline RL中clamp的问题

### ISSUE THREAD

**Issue by Dada-Cloudzxy** [ElliottYan/LUFFY#4]
您好，在PPO or GRPO中会使用torch.clamp进行重要性采样的系数限制。如果进行offline-RL，采样策略和训练策略差异过大的时候，直接被clamp掉不是直接无梯度回传了吗？

**Comment by ElliottYan** [comment]
是的，在 PPO 和 GRPO 的常见实现中 clip 操作都是通过 torch.clamp 来进行限制的，超过上下界的部分没有梯度回传。
比如可以参考 verl 官方的实现。
https://github.com/volcengine/verl/blob/main/verl/trainer/ppo/core_algos.py#L354

**Comment by ElliottYan** [comment]
哦我理解错了，我以为你说的是 on-policy 数据生成中 old policy vs new policy 的clip 操作。

在我们的 off policy loss 里 clip 是去掉的，主要因为当把 R1 的概率置为 1 时，ratio不可能大于 1，因此 clip 上下界的操作变得不对称。而且我们希望学习小概率的 action，如果 clip 这些 action 也都没有了。

**Comment by Dada-Cloudzxy** [comment]
明白了，那刚开始接近0概率的token乘以adv的时候对总体期望微乎其微，policy shape就是为了解决这个的吗？
有没有试过直接不进行重要性采样呢？感觉如果是为了学习小概率事件引入policy shape是不是和不进行重要性采样退化为SFT loss差不多呀

**Comment by Dada-Cloudzxy** [comment]
非常感谢您的回复🙏
